fs/xfstests.py

```diff
@@ -221,14 +221,22 @@ def setUp(self):
                 self.cancel("Fail to install %s required for this test." %
                             package)
         self.skip_dangerous = self.params.get('skip_dangerous', default=True)
+        self.group = self.params.get('group', default=None)
         self.test_range = self.params.get('test_range', default=None)
+        self.base_disk = self.params.get('disk', default=None)
         self.scratch_mnt = self.params.get(
             'scratch_mnt', default='/mnt/scratch')
         self.test_mnt = self.params.get('test_mnt', default='/mnt/test')
         self.disk_mnt = self.params.get('disk_mnt', default='/mnt/loop_device')
         self.fs_to_test = self.params.get('fs', default='ext4')
         self.run_type = self.params.get('run_type', default='distro')
 
+        self.devices = []
+        self.part = None
+        if (self.group and self.test_range) or (self.group is None and self.test_range is None):
```

Comment:
Regarding `(self.group is None and self.test_range is None):`

None is a bolean, it is sufficient to just check the param like `if self.group`  or `if not self.group`  and no need to explicitly use `is None` 

>>> group = None
>>> if group:
...     print(True)
... 
>>> if not group:
...     print(True)
... 
True
>>>

Reply:
any reason to have multiple same checks  ?
if (self.group and self.test_range) or (self.group is None and self.test_range is None):  

will just if (self.group and self.test_range):  not sufficient ?

Reply:
@abdhaleegit we need to have both the checks:
If both parameters are set test can not be run as its not supported together.
Also if both the parameters are none test will not run because we dont have any default value.

Reply:
@disgoel so in that case please make sure the group default value is not None, meaning

in code keep default value to 'auto' for group so the test should not cancel in case yaml is not used and also only this single check is enough in this case
if (self.group and self.test_range):

so with this
1. Always group has default auto value irrespective of yaml is provided or just .py file is run
2. (self.group is None and self.test_range is None) is not needed as at any time one param is set
3. If someone wants to run test_range.. make sure they provide yaml with empty group and test_range value, please document this in Readme

Reply:
@abdhaleegit if we keep group default value in .py with this the test_range parameter will never run because both parameters will be set and test is going to be cancelled.

Reply:
@disgoel thats the 3rd case I mentioned in previous comment, so in case of test_range test .. user must provide yaml with group as empty and a value in test_range.

basically make sure the .py work for 1 default value always ... though it will cancel fail if both values are None .. so instead make it run for default

Reply:
@PraveenPenguin your views ?

Reply:
@abdhaleegit if we keep default group, when we pass null or empty group in yaml still its going to take default value.
`self.group = self.params.get('group', default='auto')`

```
group:
# Uncomment and edit test_range for running specific tests
test_range: '1-5'

PARAMS (key=group, path=*, default=auto) => 'auto'
```

```
group: null
# Uncomment and edit test_range for running specific tests
test_range: '1-5'

PARAMS (key=group, path=*, default=auto) => 'auto'
```

Reply:
one more way we can implement is we can have empty group and test_range in yaml, user can edit for specific use cases. In .py we can pass default as none in starting and later we can check if group is none it should take default value as auto. 
Tested with this changes works fine for both group and test_range scenario.

`self.test_range = self.params.get('test_range', default=None)
`
```
            if not self.group:
                self.group = 'auto'
            cmd = './check %s -g %s' % (args, self.group)
```
@abdhaleegit @PraveenPenguin your thoughts with this approach ?

Reply:
> one more way we can implement is we can have empty group and test_range in yaml, user can edit for specific use cases. In .py we can pass default as none in starting and later we can check if group is none it should take default value as auto. Tested with this changes works fine for both group and test_range scenario.
> 
> `self.test_range = self.params.get('test_range', default=None) `
> 
> ```
>             if not self.group:
>                 self.group = 'auto'

so this would add one extra check.. but the same can be achieved by
self.test_range = self.params.get('test_range', default=auto)

Reply:
> No, yaml input value takes precedence over py default values
> Also do not use null, instead use None with out qoutes

I have set group default value as auto in py and None in yaml. Its taking None as a group.

```
group: None
PARAMS (key=group, path=*, default=auto) => 'None'
PARAMS (key=test_range, path=*, default=None) => '1-5'
Running './check  -E /var/tmp/avocado_lt6n6xnw/exclude -g None'
```

Reply:
I agree with @abdhaleegit view, the test should run without ymal and ymal has high precedence always , Rest looks good to me

Reply:
@PraveenPenguin @abdhaleegit have addressed the changes, please review. Also updated the README file with latest changes.